agent/node.go

```diff
@@ -607,8 +607,9 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			return err
 		}
 		done := make(chan struct{})
+		var runErr error
 		go func() {
-			m.Run(context.Background()) // todo: store error
+			runErr = m.Run(context.Background())
```

Comment:
@stevvooe That is because in contrast to agent and node, for the manager cancelling the context and calling `manager.Stop` do not behave the same way. Or at least didn't when this was written, I see manager calls `Stop` now multiple places what may work if all the cases are covered but opposite to what is happening in node and agent. So to safely quit the manager we call `Stop()` on it and then wait for the `Run()` method to return.